selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/stacks/StackDetails.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.pageobject.dashboard.stacks;
+
+import static java.lang.String.format;
+
+import com.google.inject.Inject;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.utils.WaitUtils;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.PageFactory;
+
+public class StackDetails {
+  private final SeleniumWebDriverHelper seleniumWebDriverHelper;
+
+  @Inject
+  public StackDetails(
+      SeleniumWebDriver seleniumWebDriver, SeleniumWebDriverHelper seleniumWebDriverHelper) {
+    this.seleniumWebDriverHelper = seleniumWebDriverHelper;
+    PageFactory.initElements(seleniumWebDriver, this);
+  }
+
+  private interface Locators {
+    String TOOLBAR_XPATH_PATTERN = "//div[@che-title='%s']";
+    String NEW_STACK_NAME = "deskname";
+    String SAVE_CHANGES_BUTTON_NAME = "saveButton";
+    String TO_ALL_STACKS_LIST_BUTTON_XPATH = "//a[@title='All stacks']";
+  }
+
+  public void waitToolbar(String stackName) {
+    seleniumWebDriverHelper.waitVisibility(
+        By.xpath(format(Locators.TOOLBAR_XPATH_PATTERN, stackName)));
+  }
+
+  public void setStackName(String stackName) {
+    WebElement stackNameField =
+        seleniumWebDriverHelper.waitVisibility(By.name(Locators.NEW_STACK_NAME));
+    stackNameField.click();
```

Comment:
We can replace to commands here by one: `seleniumWebDriverHelper.waitAndClick()`